fe/parameterize_batch.py

```diff
@@ -0,0 +1,387 @@
+            true_dG_list.append(true_dG)
+
+    if args.precision == 'single':
+        precision = np.float32
+    elif args.precision == 'double':
+        precision = np.float64
+    else:
+        raise Exception("precision must be either single or double")
+
+    num_gpus = args.num_gpus
+    num_workers = args.num_gpus*args.jobs_per_gpu
+
+    print('Creating multiprocessing pool with',args.num_gpus, 'gpus and', args.jobs_per_gpu, 'jobs per gpu')
+    pool = multiprocessing.Pool(num_workers)
+
+    start = 1e3
+    end = 1.0
+    NT = 500
+    base = np.exp(np.log(end/start)/NT)
+    exps = np.arange(NT)
+    part_one = np.power(base, exps)*start
+    part_two = np.linspace(1.0, 0.3, 1000)
+    part_three = np.linspace(0.3, 0.0, 4000)
+
+    forward_schedule = np.concatenate([part_one, part_two, part_three])
+    backward_schedule = forward_schedule[::-1]
+    lambda_schedule = np.concatenate([forward_schedule, backward_schedule])
+
+    T = lambda_schedule.shape[0]
+    assert T % 2 == 0
+    dt = 0.001
+    step_sizes = np.ones(T)*dt
+    #T_min = 500
+    assert T % 2 == 0
+    cas = np.ones(T)*0.93
+    epoch = 0
+    lr = 5e-4
+
+    ligand_list = []
+    for ligand_sdf_file in args.ligand_sdf_file:
+        f = open(ligand_sdf_file, 'r').readlines()
+        for file_name in f:
+            ligand_list.append(Chem.SDMolSupplier(file_name.rstrip('\n'), removeHs=False))
+
+    for guest_mol in ligand_list[0]:
+        break
+    
+    off = smirnoff.ForceField("test_forcefields/smirnoff99Frosst.offxml")
+    guest_potentials, (guest_params, guest_param_groups), masses = forcefield.parameterize(guest_mol, off, dimension=4)
+    
+    opt_init, opt_update, get_params = optimizers.adam(lr)
+    opt_state = opt_init(guest_params)
+
+    np.savetxt("original_off_params.txt", guest_params)
+   
+    host_pdb_file = args.complex_pdb
+    host_pdb = app.PDBFile(host_pdb_file)
+    host_conf = []
+    for x,y,z in host_pdb.positions:
+        host_conf.append([to_md_units(x),to_md_units(y),to_md_units(z)])
+    host_conf = np.array(host_conf)
+    host_name = "complex"  
+
+    #initial_params = sim.combined_params
+    #opt_init, opt_update, get_params = optimizers.adam(lr)
+    #opt_state = opt_init(initial_params)
+
+    num_epochs = 100
+    # torsion, charge, gb radii, gb scaling factor, torsion
+    vary_allowed_groups = [4]*10 + [17]*20 + [18]*20 + [19]*20 + [4]*30
+
+    assert len(vary_allowed_groups) == num_epochs
+
+    ligand_dataset = dataset.Dataset(ligand_list)
+    ligand_dataset.iterbatches(1)
+    train, test = ligand_dataset.split(0.7)
+
+    for epoch in range(num_epochs):
+        # sample from the rdkit DG distribution (this can be changed later to another distribution later on)
+        # epoch_params = get_params(opt_state)
+        # guest_params = epoch_params
+
+        train.shuffle()
+        print(train.data[0])
+
+###### Training ######
+        for i in range(len(train.data)):
```

Comment:
loop over the data using iterbatches(i), this was designed initially to iterate over any batch size (including 1).

```
for data in train.iterbatches(i):
      guest_mol = data[0]
```

Reply:
I changed both training and test accordingly.